Microsoft.NET.Build.Containers/LocalDocker.cs

```diff
@@ -59,8 +59,21 @@ public static async Task WriteImageToStream(Image x, string name, string tag, St
             }
             else
             {
-                throw new NotImplementedException("Need a good error for 'couldn't download a thing because no link to registry'");
-            }        }
+                // try a last-chance load from local if no registry present
+                string localPath = ContentStore.PathForDescriptor(d);
+                if (File.Exists(localPath))
+                {
+                    // Assume file is up to date and just return it
+                    string layerTarballPath = $"{d.Digest.Substring("sha256:".Length)}/layer.tar";
+                    await writer.WriteEntryAsync(localPath, layerTarballPath);
+                    layerTarballPaths.Add(layerTarballPath);
+                }
+                else
+                {
+                    throw new NotImplementedException("Need a good error for 'couldn't download a thing because no link to registry'");
+                }
```

Comment:
This was needed because some layers may not exist on a registry. In fact, our publish-dir layers go through the first pathway (but I don't think I like that) - I'd like to not use the `DownloadBlob` method to do double duty there.